Make sure we run logs versus standard indexing

salvatore-campagna opened the pull request:
Before introducing this fix we were running both data streams using LogsDB as a result of logs templates being used.
Here we make sure we use LogsDB only for the contender. We also fix a mapping issue which results in using a text field when fields are dynamically mapped. Having the `host.name` mapped as a text field prevents us from running aggregations
due to missing doc values.

salvatore-campagna committed:
fix: make sure we run logs versus standard indexing

Before introducing this fix we were running both data streams using LogsDB
as a result of logs templates being used. Here we make sure we use LogsDB
only for the contender. We also fix a mapping issue which results in using
a text field when fields are dynamically mapped.

jan-elastic committed:
Disable tests when AdaptiveAllocationsFeatureFlag is disabled (#110969)

elasticsearchmachine commented:
Pinging @elastic/es-storage-engine (Team:StorageEngine)

salvatore-campagna committed:
Update docs/changelog/110973.yaml

elasticsearchmachine commented:
Hi @salvatore-campagna, I've created a changelog YAML for you.

salvatore-campagna committed:
checkstyle: note missing space

original-brownbear committed:
Optimize transport_thread exectures search logic (#110965)

We can save a bunch of allocations and cycles in a few spots, the blocks
checks in particular are very heavy if done in a loop over many indices.
Also, presizing some collections, avoiding temporary collections where
not necessary etc. helps a bit as well. None of this is very relevant
for searches over few indices but all of it becomes quite visible in
profiling once you target a larger number of indices/shards.

not-napoleon committed:
Add in format nanos function (#110934)

Adds a helper function to format a single long representing nanoseconds
since epoch.  I found myself writing this in a few places while working
on https://github.com/elastic/elasticsearch/issues/109987 and decided it
would be a good helper method.

ywangd committed:
[Test] Test for EOFException when skipping beyond available data (#110953)

See title

alex-spies committed:
ESQL: Correctly compute Rename's output (#110968)

Calling Rename.output() previously returned wrong results.

Since #110488, instead it throws an IllegalStateException. That leads to test failures in the EsqlNodeSubclassTests because e.g. MvExpandExec and FieldExtractExec eagerly calls .output() on its child when it's being constructed, and the child can be a fragment containing a Rename.

salvatore-campagna commented:
The test failing here

```
./gradlew ":modules:data-streams:javaRestTest" --tests "org.elasticsearch.datastreams.logsdb.qa.StandardVersusLogsIndexModeChallengeRestIT.testTermsQuery" -Dtests.seed=7912BAA000049E5F -Dtests.locale=fr-CA -Dtests.timezone=Europe/Lisbon -Druntime.java=22
```

is failing because it needs the `host.name` field being injected which is done by #110938.
I will merge that PR first and then make sure this works.

This happens in case dynamic mapping is used by the contender where the `host.name` is not included in the mapping and sorting fails because of the missing field.

arteam committed:
Update comments in IndexShardTests#testScheduledRefresh (#110943)

* First scheduledRefresh returns false because search is idle
* Remove the comment about the inability to control the result of scheduleRefresh

Follow-up for #110312

nik9000 committed:
ESQL: Expand main javadoc (#110941)

This expands the package level javadoc for ESQL to reference a few more
things and include some of our design rules of thumb. It also breaks it
out into more different sections to line up better with including more
stuff.

nik9000 committed:
AwaitsFix: https://github.com/elastic/elasticsearch/issues/110977

original-brownbear committed:
Further speedups to can-match (#110909)

Most importantly this one removes spurious synchronization.
Also, saves some more allocations and indirections in another sorting
spot.

lkts commented on docs/changelog/110973.yaml:
```diff
@@ -0,0 +1,5 @@
+pr: 110973
+summary: Make sure we run logs versus standard indexing
+area: Logs
+type: bug
```
I don't think we should mark this as bug since it will go to the changelog.

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT #110978

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests testInfoParameters {class org.elasticsearch.xpack.esql.plan.physical.MvExpandExec} #110980

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.tree.EsqlNodeSubclassTests testInfoParameters {class org.elasticsearch.xpack.esql.plan.physical.FieldExtractExec} #110981

salvatore-campagna committed:
Delete docs/changelog/110973.yaml

martijnvg committed:
Adding minimal docs around using index mode logs. (#110932)

This adds minimal docs  around how to the new logs index mode for data
streams (most common use case). This is minimal because logs index mode
is still in tech preview. Minimal docs should allow any interested users
to experiment with the new logs index mode.

not-napoleon committed:
[ESQL] string switch in estimate size (#110946)

Move the estimated size out of EstimateRowSize and onto DataType.  This eliminates a string switch, and makes it clear that writers of new data types should think about the size when possible.

benwtrent committed:
Make empty string searches be consistent with case (in)sensitivity (#110833)

If we determine that the searchable term is completely empty, we switch back to a regular term query. This way we return the same docs as expected when we do a case sensitive search.

closes: #108968

kderusso committed:
Fix references to incorrect query rule criteria type (#110994)

DiannaHohensee committed:
Add snapshot code comments and couple renames (#110613)

Renamed SnapshotsInProgress.Entry.shardsByRepoShardId()
and ShardGeneration.toBlobNamePart(), updated some variable
names, and generally added/clarified comments throughout
the snapshotting code.

joegallo committed:
Directly download commercial ip geolocation databases from providers (#110844)

Co-authored-by: Keith Massey <keith.massey@elastic.co>

elasticsearchmachine committed:
Mute org.elasticsearch.search.sort.FieldSortIT testIssue6614 #110999

ywangd committed:
Add support for shared input stream to RangeMissingHandler  (#110855)

This PR augments the RangeMissingHandler interface to support shared
input stream which is reused when filling mulitple gaps. The shared
input stream is meant to be consumed sequentially to fill the list of
gaps in sequential order. The existing behaviour is preserved when
shared input stream is not used, i.e. when it is `null`.

DaveCTurner committed:
Remove unused `AsyncSnapshotInfo#getSnapshotId` (#111005)

This method became unused with #110957.

leemthompo committed:
[DOCS] Retrievers and rerankers (#110007)

Co-authored-by: Adam Demjen <demjened@gmail.com>

tvernum committed:
Update logging for role mappings (#110870)

Fixes the formatter for log messages in ExpressionRoleMapping and adds
more logging in ClusterStateRoleMapper and NativeRoleMappingStore

salvatore-campagna committed:
Extract method and remove unused variable (#110975)

carlosdelest committed:
Clarify synonyms docs (#110822)

alex-spies committed:
Unmute EsqlNodeSubclassTests (#110985)

The test failure was already fixed in
https://github.com/elastic/elasticsearch/pull/110968.

DaveCTurner committed:
Combine per-repo results in get-snapshots action (#111004)

With #107191 we can now safely accumulate results from all targetted
repositories as they're built, rather than staging each repository's
results in intermediate lists in case of failure.

DaveCTurner committed:
Add docs on threading in get-snapshots action (#111006)

The threading here is kinda subtle these days. This commit adds some
more docs and assertions in this area.

thecoop committed:
Update known-issues for the features upgrade bug, and increase scope to include 8.12.x (#111014)

salvatore-campagna committed:
Inject `host.name` field without relying on (component) templates (#110938)

We do not want to rely on templates or component templates to include
the host.name field in indices using LogsDB. The host.name field is a field
we sort on by default when LogsDB is used. As a result, we just inject it
by default, the same way we do for the @timestamp field. This prevents
sorting errors due to missing host.name field in mappings.

The host.name is a keyword field and depending on the value of subobjects it will
be mapped as a name keyword nested inside a host or as a flat host.name keyword.
We also include ignore_above as we normally do for keywords in observability mappings.

nicktindall committed:
Use enums instead of booleans in TransportReplicationAction (#111017)

Relates to: #109414

DaveCTurner committed:
Always allow rebalancing by default (#111015)

Today `cluster.routing.allocation.allow_rebalance` defaults to
`indices_all_active` which blocks all rebalancing moves while the
cluster is in `yellow` or `red` health. This was appropriate for the
legacy allocator which might do too many rebalancing moves otherwise.
The desired-balance allocator has better support for rebalancing a
cluster that is not in `green` health, and expects to be able to
rebalance some shards away from over-full nodes to avoid allocating
shards to undesirable locations in the first place. This commit changes
the default `allow_rebalance` setting to `always`.

tteofili committed:
Dense vector field types updatable for int4 (#110928)

elasticsearchmachine committed:
Mute org.elasticsearch.ingest.geoip.EnterpriseGeoIpDownloaderIT testEnterpriseDownloaderTask #111002

leemthompo committed:
[DOCS] Fix rendering bug (#111025)

Closes https://github.com/elastic/elasticsearch/issues/111023

elasticsearchmachine committed:
Mute org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT test {p0=data_stream/190_failure_store_redirection/Ensure failure is redirected to correct failure store after a reroute processor} #111041

original-brownbear committed:
Fix Comparator returned by MinMax (#111029)

The `right == null` case was obviously wrong here for sorting nulls at
either end.

closes #110991

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests org.elasticsearch.xpack.esql.querydsl.query.SingleValueQueryTests #111042

DaveCTurner committed:
Collapse nested listeners in get-snapshots action (#111028)

There's no need for two layers of `RefCountingListener` any more, we
propagate all failures to the top level in any case, so we can run
everything under a single listener.

elasticsearchmachine committed:
Mute org.elasticsearch.compute.lucene.ValueSourceReaderTypeConversionTests testLoadAllStatusAllInOnePage #111048

benwtrent committed:
Clean up some of the kNN code, removing unused paths (#110972)

These `float[]` functions were only used by tests. Instead we should use
the `VectorData` abstraction.

Additionally, this updates our internal abstractions to take advantage
of our vector index type enumerations. This unifies and simplifies
validation logic, etc.

ioanatia committed:
[ESQL] Minor refactor for DataType (#111016)

* Remove duplicate functions from EsqlDataTypes

* Remove unused methods from DataType

ezimuel committed:
Changed security API endpoints to stable (#110862)

benwtrent committed:
Unmuting #110999 after #111029 (#111052)

this is now fixed after #111029 

closes https://github.com/elastic/elasticsearch/issues/110999

iverase committed:
[ESQL] Run SingleValueMatchQuery approximation last (#111039)

Currently we run SingleValueQueries as a filter composed of the original
and a doc values queries that filter multivalues. It might happen that
the latter drives the iteration resulting in warnings of multivalue
fields even if the query does not match any. This PR proposes to
increase the match cost for the doc values query so it always run last.
This makes it easier to test and it should never give false warnings.

fixes https://github.com/elastic/elasticsearch/issues/110977

iverase committed:
Speed up collecting zero document string terms (#110922)

Use segment ordinals when possible to collect zero document buckets

naj-h committed:
Make skip_unavailable=true a notable change instead of breaking change (#110983)

nielsbauman committed:
Fix allowed warning during template creation in YAML test (#111059)

Fixes #111041

alex-spies committed:
ESQL: Skip retrofitted tests (#111019)

prwhelan committed:
[Transform] Allow task canceling of validate API calls (#110951)

Validate will initiate a search request.  In the event that the search
request needs to be cancelled, rather than manually stopping the task,
cancelling the Validate task will now propagate the cancel request to
the Search task.

Relate #88010

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticsearchmachine committed:
Mute org.elasticsearch.action.search.KnnSearchSingleNodeTests testKnnSearchAction #111072

stefnestor committed:
(Doc+) Flush out Data Tiers (#107981)

I highly value the content on this [Data Tiers](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html) page. Thanks for writing it! In my experience, some users may become slightly confused by its golden nuggets due to its brevity. This PR attempts to flush out common questions while remaining concise. 

The main changes are in the first and second-to-last sections; however, I do attempt some heading restructuring to make the TOC idea-groupings more clear for easier scan-throughs. 

The specific clarifications I'd like to push in order of appearance:

- There's content tier (for "data category" > "content" as we've dubbed it on the higher page) and the data temperature tiers (for time series). That the temperature tiers group together is technically not stated so users end up asking about when they'd go hot>warm vs content>warm, etc. I suspect this confusion is only because users come straight to this page instead of starting at the hierarchy-parent page so have linked up. 
- (Main) Frozen being accessed/searched "rarely" should imply, well rarely. I wrote 1% in the PR `[TIP]` guideline section as a discussion starting point. Frequently we see users not understanding either that they actually have been or that they shouldn't have ≥25% of all searches hitting frozen tier. This comes up because of architecture bugs (e.g. frozen indices with future timestamps) but also just happenstance (e.g. 01605242 where of searches they hit majority hot, ~5% cold, but then again hit 75% frozen).
- There's a slew of "how do I check that?", "how do I change that (at creation/later)?", "what if I set it null?" questions we get about `_tier_preference` so just extended the existing section already about it. 

---------

Co-authored-by: shainaraskas <58563081+shainaraskas@users.noreply.github.com>
Co-authored-by: Lee Hinman <dakrone@users.noreply.github.com>

elasticsearchmachine committed:
Mute org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapperTests testKnnQuantizedFlatVectorsFormat #111044

joegallo committed:
Add an 8.15 backport version for the ENTERPRISE_GEOIP_DOWNLOADER (#111069)

ywangd committed:
Add BytesCountingFilterInputStream for counting bytes read (#111020)

See title

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoPointIT testPushedDownQueriesSingleValue #111084

salvatore-campagna committed:
test: re-enable test after backport #11031 (#111035)

kingherc committed:
Correct force merge disk space requirements (#111066)

Correct force merge disk space requirements

luigidellaquila committed:
ES|QL: more strict management of NOW (#111085)

Making the management of NOW (current time) more strict. Now we assume
that it's always available in the Configuration (as it practically was
already, apart from some mocks in tests).

original-brownbear committed:
Cleanup some ActionListener use in SearchTransportService (#111067)

* remove publicly exposing the `SearchActionListener` outside of its
  module scope
* use constants for the free response and the noop transport listener
  to remove some pointless transport thread allocations

ivancea committed:
ESQL: Fix Double operations returning infinite (#111064)

Fixes Add, Sub and Mul to return a null and a warning (using `warnExceptions`).

Partially fixes https://github.com/elastic/elasticsearch/issues/111026

salvatore-campagna committed:
Rename `logs` index mode to `logsdb` (#111054)

breskeby committed:
Update Gradle wrapper to 8.9 (#110109)

nik9000 committed:
ESQL: Check for circuit breaker in async (#111070)

This adds a test to make sure that circuit breakers exceptions thrown
from the async ESQL task are properly recorded to the index.

DaveCTurner committed:
Introduce `EnumSerializationTestUtils` (#111092)

Adds a test utility for fixing the ordinal-based wire protocol of enums.

Relates #107944

ivancea committed:
Added Percentile aggregation tests and Kibana docs (#111050)

- Added Percentile aggregation tests and autogen docs
- Added a new "appendix" section to FunctionInfo. Existing Percentile docs had a final, long section with info, and we need this to leep it. We have an "detailedDescription" attribute already, but it's right after the description, and it would make it harder to read the important bits of the function (types, examples...). So I'm not reusing it.

rjernst committed:
Skip preallocate tests on windows (#110998)

The preallocate tests assumed that preallocation was using the fallback
implementation which calls setLength on Windows. However, that fallback
only happens inside the SharedBytes class, so windows doesn't actually
do anything when tryPreallocate is called. This commit skips the test on
windows.

closes #110948

idegtiarenko committed:
Update ShardChangesObserver logging (#110748)

This change adds logging for the moment of time when shard starts initializing (for the cases where initialization duration matters) as well as the initialization source (as PEER/EMPTY_STORE/etc)

masseyke committed:
Adding mapping validation to the simulate ingest API (#110606)

ivancea committed:
ESQL: Support IP fields in MAX and MIN aggregations (#110921)

- Support IP in MAX() and MIN()
  - Used a custom IpArrayState for it, as it's quite different from the `X-ArrayState.java.st` generated ones
- Add IP test cases for aggregation tests

rjernst committed:
Use ESTestCase for vector sysprop tests (#110990)

The VectorSystemPropertyTests need to run a child process. Normally this
isn't possible since we run with security manager, but the
`@WithoutSecurityManager` annotation causes the test suite to run with
the security manager disabled. However, that annotation only works with
ESTestCase. This commit changes the base class of the test suite to
ESTestCase.

closes #110949

breskeby committed:
Make --write-verification-metadata faster (#111103)

by ignoring elasticsearch distributions when resolving artifacts

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT test {stats.MaxOfIpGrouping} #111107

breskeby committed:
Cleanup repository after adding accidental files to PR (#111108)

With https://github.com/elastic/elasticsearch/pull/110109 i accidentally added local files to the repo.
This removes them again.

masseyke committed:
Fixing EnterpriseGeoIpDownloaderIT (#111080)

not-napoleon committed:
[ESQL] Top level javadoc outlining how the planner works. (#111075)

Expanding the overview java doc to explain the steps involved in planning.
---------

Co-authored-by: Alexander Spies <alexander.spies@elastic.co>

rjernst committed:
Add Java 21 to archive toolchain resolver (#111112)

Since the bundled JDK is now Java 22, the archive resolver needs to
support resolving Java 21. This commit updates the resolver to know
which version of Java 21 to download.

parkertimmins committed:
PUT slm policy should only increase version if actually changed (#111079)

Currently, repeated PUTs to PUT /_slm/policy/ with the same policyId and policy definition will increase the policy version and update the modified_date. This will cause problems for interval based SLM scheduling as the modified_date is used as the start time for scheduling intervals. If the policy modified_date is updated repeatedly, this could cause an interval based policy to never run. This is fine for manual policy updates, but repeated policy PUTs in a loop due to automation could be a problem.
---------
Co-authored-by: Joe Gallo <joegallo@gmail.com>

elasticsearchmachine committed:
Mute org.elasticsearch.ingest.geoip.EnterpriseGeoIpDownloaderIT testEnterpriseDownloaderTask #111002

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.spatial.SpatialPushDownCartesianPointIT testPushedDownQueriesSingleValue #110982

rjernst committed:
Allow runtime java to match adoptium jdks (#111117)

This commit relaxes the toolchain specification when runtime.java is set
to allow the adoptium toolchain resolver to match the jdk. Before this
only the oracle toolchain resolvers could match. It also removes LTS
releases from the archived oracle resolver since those are not EOL.

mhl-b committed:
Use NodesStatsRequestParameters.Metric enums more over string (#110810)

Use EnumSet for NodesStatsRequestParameters.Metric

---------

Co-authored-by: David Turner <david.turner@elastic.co>

elasticsearchmachine committed:
Mute org.elasticsearch.multi_node.GlobalCheckpointSyncActionIT org.elasticsearch.multi_node.GlobalCheckpointSyncActionIT #111124

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT test {stats.MinOfIpGrouping} #111125

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.security.authz.store.NativePrivilegeStoreCacheTests testGetPrivilegesUsesCache #110788

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.inference.rank.textsimilarity.TextSimilarityRankTests testRerankInferenceResultMismatch #111133

elasticsearchmachine committed:
Mute org.elasticsearch.cluster.PrevalidateShardPathIT testCheckShards #111134

masseyke committed:
Making sure that watcher-history index is green in HistoryIntegrationTests (#111119)

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT test {stats.MaxOfIpGrouping SYNC} #111136

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT test {stats.MinOfIpGrouping SYNC} #111137

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT test {stats.MinOfIpGrouping ASYNC} #111138

elasticsearchmachine committed:
Mute org.elasticsearch.packaging.test.DockerTests test021InstallPlugin #110343

elasticsearchmachine committed:
Mute org.elasticsearch.xpack.restart.FullClusterRestartIT testDisableFieldNameField {cluster=UPGRADED} #111141

Philippus committed:
Add comma before charset parameter in `WWW-Authenticate` response header (#110906)

Add comma before charset parameter in WWW-Authenticate response header,
according to RFC 7617.

idegtiarenko committed:
Return only requested master stats (#110970)

DaveCTurner committed:
Rework docs on logging levels (#111143)

Clarify that the default config is the recommended one, and that users
should not normally enable `DEBUG` or `TRACE` logging without looking at
the source code. Also reorders the information a bit for easier reading.

shane-zfx committed:
Optimize loop in URL decoding #110235 (#110237)

when the target character is detected, the loop ends
---------
Co-authored-by: shane-zfx <zffxxx7@gmail.com>

ivancea committed:
Added Sum aggregation tests and docs (#110984)

- Added SUM() agg tests (Which autogenerates docs)
- Converted non-finite doubles to nulls in aggregator

The complete set of tests depends on
https://github.com/elastic/elasticsearch/issues/110437, as commented in
code. After completion, the test can be uncommented and everything
should work fine

original-brownbear committed:
Safely ensure search contexts are release in ESSingleNodeTestCase (#111153)

The transport action for freeing contexts is forking to GENERIC
nowadays and executed in a fire-and-forget manner in production code.
When it was running on the direct executor, no busy wait was needed for
a single node since the freeing was fully synchronous inside the
searches.
With forking, the waiting on free tasks does not help because the task
is created inside the forked execution for a local node execution, so we
need to busy wait like we do for multi-node clusters here too.
-> dry up the code and use the same method for both.

ivancea committed:
ESQL: TOP aggregation IP support (#111105)

Added IP support to TOP() aggregation.

Adapted a bit the stringtemplates organization for esql/compute to
(also?) work with specific datatypes. Right now it may be a bit messy,
but we need the specific support for cases like this.

ivancea committed:
ESQL: Unmute and fix Max/Min CSV tests with unsorted IPs (#111149)

Fixes https://github.com/elastic/elasticsearch/issues/111107
Fixes https://github.com/elastic/elasticsearch/issues/111125
Fixes https://github.com/elastic/elasticsearch/issues/111136
Fixes https://github.com/elastic/elasticsearch/issues/111137
Fixes https://github.com/elastic/elasticsearch/issues/111138

ivancea committed:
ESQL: Enhaced docs on muting ESQL CSV tests (#111152)

Added some docs around the possibilities to mute all tests for a csv
case.

Closes https://github.com/elastic/elasticsearch/issues/109664

stefnestor committed:
Add link to Max Shards Per Node docs (#110993)

... from exception message

nik9000 committed:
ESQL: Move a test util out of production code (#111163)

This moves a testing utility out of the production code path. It had to
be in the production path a while ago, but that's since been fixed and
now it really belongs under src/test/java.

rjernst committed:
Use bundled jdk version for immutable collections patch (#111162)

When building our patch for immutable collections we must find a jdk to
use for the build. If runtime java is set, this is clear. But if there
is no runtime java, we intend to fallback to the bundled jdk version.
Yet the runtime version in this case does not itself fallback to the
bundled jdk. This commit changes the immutable collections patch build
to explicitly use the bundled jdk version.

masseyke committed:
Avoid calling real maxmind endpoint from EnterpriseGeoIpDownloader #111120 (#111121)

If the geoip_use_service system property is set to true, then the
EnterpriseGeoIpHttpFixture is disabled, so EnterpriseGeoIpDownloader
incorrectly calls maxmind.com without credentials, and fails. This
change makes it so that the maxmind server is never called from tests.
Closes https://github.com/elastic/elasticsearch/issues/111002

mark-vieira committed:
Unmute DockerTests.test600Interrupt (#111165)

Investigating https://github.com/elastic/elasticsearch/issues/111132 and
it seems this test has been muted on `main` for some time. Let's unmute,
to see if this is specific to the 7.17 branch or not.

nik9000 committed:
ESQL: Fix test failure around field loading (#111058)

Bumps the time limit. Sometimes the test data is big
enough to warrant more time. And CI isn't always fast.
Closes #111048
Closes #110244

fang-xing-esql committed:
[ES|QL] Generate docs for unregistered esql functions from annotations (#108749)

* render docs for operators

lkts committed:
Fix remaining references to logs index mode (#111164)

nik9000 committed:
ESQL: Reenable rarely failing test (#111172)

This test rarely fails and never locally. I don't have any information
about *why*. This adds extra debugging and reenables the test. If it
fails again we'll have more information.

Closes #111141

nik9000 committed:
ESQL: Allow suppressed exeption in HeapAttack (#111173)

We're ok with suppresses exceptions in this test.

Closes #111128

kanoshiou committed:
Fix unnecessary mustache template evaluation (#110986)

Addresses the performance issue in the date ingest processor where Mustache template evaluation is unnecessarily applied inside a loop. The timezone and locale templates are now evaluated once before the loop, improving efficiency.

closes #110191
---------
Co-authored-by: Joe Gallo <joegallo@gmail.com>

elasticsearchmachine committed:
Mute org.elasticsearch.packaging.test.DockerTests test600Interrupt #111132

tteofili committed:
Document how to update dense vector field type (#111038)

salvatore-campagna committed:
fix: hostname injected automatically
